tests/integration/test_network_configuration.py

```diff
@@ -0,0 +1,248 @@
+            },
+        )
+
+        def cleanup():
+            bucket = s3_resource.Bucket(bucket_name)
+            bucket.objects.all().delete()
+            bucket.object_versions.all().delete()
+            bucket.delete()
+
+        cleanups.append(cleanup)
+
+        assert_host_customisation(res["Location"], use_hostname_external=True)
+
+    def test_multipart_upload(self, s3_bucket, s3_client, assert_host_customisation):
+        key_name = f"key-{short_uid()}"
+        upload_id = s3_client.create_multipart_upload(Bucket=s3_bucket, Key=key_name)["UploadId"]
+        part_etag = s3_client.upload_part(
+            Bucket=s3_bucket, Key=key_name, Body=b"bytes", PartNumber=1, UploadId=upload_id
+        )["ETag"]
+        res = s3_client.complete_multipart_upload(
+            Bucket=s3_bucket,
+            Key=key_name,
+            MultipartUpload={"Parts": [{"ETag": part_etag, "PartNumber": 1}]},
+            UploadId=upload_id,
+        )
+
+        assert_host_customisation(res["Location"], use_hostname_external=True)
+
+    def test_201_response(self, s3_bucket, s3_client, assert_host_customisation):
+        key_name = f"key-{short_uid()}"
+        body = "body"
+        presigned_request = s3_client.generate_presigned_post(
+            Bucket=s3_bucket,
+            Key=key_name,
+            Fields={"success_action_status": "201"},
+            Conditions=[{"bucket": s3_bucket}, ["eq", "$success_action_status", "201"]],
+        )
+        files = {"file": ("my-file", body)}
+        res = requests.post(
+            presigned_request["url"],
+            data=presigned_request["fields"],
+            files=files,
+            verify=False,
+        )
+        res.raise_for_status()
+        json_response = xmltodict.parse(res.content)["PostResponse"]
+
+        assert_host_customisation(json_response["Location"], use_hostname_external=True)
+
+
+class TestSQS:
+    """
+    Test all combinations of:
+
+    * endpoint_strategy
+    * sqs_port_external
+    * hostname_external
+    """
+
+    def test_off_strategy_without_external_port(
+        self, monkeypatch, sqs_create_queue, assert_host_customisation
+    ):
+        monkeypatch.setattr(config, "SQS_ENDPOINT_STRATEGY", "off")
+
+        queue_name = f"queue-{short_uid()}"
+        queue_url = sqs_create_queue(QueueName=queue_name)
+
+        assert_host_customisation(queue_url, use_localhost=True)
+        assert queue_name in queue_url
+
+    def test_off_strategy_with_external_port(
+        self, monkeypatch, sqs_create_queue, assert_host_customisation
+    ):
+        external_port = 12345
+        monkeypatch.setattr(config, "SQS_ENDPOINT_STRATEGY", "off")
+        monkeypatch.setattr(config, "SQS_PORT_EXTERNAL", external_port)
+
+        queue_name = f"queue-{short_uid()}"
+        queue_url = sqs_create_queue(QueueName=queue_name)
+
+        assert_host_customisation(queue_url, use_hostname_external=True)
+        assert queue_name in queue_url
+
```

Comment:
for completeness' sake:
`assert f":{external_port}" in queue_url`
Since this is the one case where the port actually changes due to `SQS_PORT_EXTERNAL`